### PR TITLE
ACK or NACK messages

### PR DESCRIPTION
**Pull request by valnoel**
Fix issue #4

### CODE REVIEW

File: src/connection.py [hunk 1]
```diff
@@ -52,14 +52,20 @@ def connect(self, queues):
...
+                      no_ack=no_ack)
 
         logging.info('Service started, waiting messages ...')
         self.channel.start_consuming()
 
+    def acknowledge_message(self, delivery_tag):
+        self.channel.basic_ack(delivery_tag)
+
+    def reject_message(self, delivery_tag):
+        self.channel.basic_nack(delivery_tag)
```
Comment (MarcAntoine-Arnaud) [inline src/connection.py]: instead of creating 2 methods for that, you can simply return a boolean.
if == True => ACK, else NACK.

Comment (MarcAntoine-Arnaud) [reply src/connection.py]: don't miss to catch also exception and nack if you have some.

Comment (valnoel) [reply src/connection.py]: OK, see: e775936 and 2ce735d

File: src/connection.py [hunk 2]
```diff
@@ -60,11 +60,11 @@ def consume(self, queue, callback, no_ack=True):
         logging.info('Service started, waiting messages ...')
         self.channel.start_consuming()
 
-    def acknowledge_message(self, delivery_tag):
-        self.channel.basic_ack(delivery_tag)
-
-    def reject_message(self, delivery_tag):
-        self.channel.basic_nack(delivery_tag)
+    def acknowledge_message(self, delivery_tag, ack=True):
```
[the remark this answers is not in the document]
Comment (valnoel) [reply src/connection.py]: Understood! See 4ed01ee274b1a01e2b2686e052963150ad8e29d7 :wink:

File: src/connection.py [hunk 3]
```diff
@@ -5,6 +5,23 @@
...
+        self.channel = channel
+
+    def callback(self, ch, method, properties, body):
+        ack = False
+        try:
+            ack = self.consumer_callback.__call__(ch, method, properties, body)
+        except Exception as e:
+            logging.error("An error occurred in consumer callback: %s", e)
+
+        if ack is None or ack is True:
```
Comment (MarcAntoine-Arnaud) [inline src/connection.py]:
```python
if ack in [None, True]:
```

Comment (valnoel) [reply src/connection.py]: Nice! Fixed in 2db82b4!

**Pull request merged**